pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -390,13 +397,16 @@ def training_step(self, *args, **kwargs):
         return self.model(*args, **kwargs)
 
     def validation_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        if isinstance(self.model, DistributedDataParallel):
+            return self.model(*args, **kwargs)
+        else:
+            return self.lightning_module.validation_step(*args, **kwargs)
 
     def test_step(self, *args, **kwargs):
-        return self.model(*args, **kwargs)
+        return self.lightning_module.test_step(*args, **kwargs)
```

Comment:
Could this be an issue for the plugins sub-classing DDP or DDPSpawn, like DeepSpeed ?

Reply:
Hi, @tchaton, sorry about the confusion. this PR should only focus on prepare_data_per_node. I've rebased this PR and submit again to make it clean.